File: CS/testSample/TableBarItem/CustomController/MyBarAndDockingController.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraBars;
using DevExpress.XtraBars.Styles;
using DevExpress.XtraBars.Controls;
using DevExpress.XtraEditors;
using DevExpress.Utils;
using DevExpress.XtraBars.ViewInfo;

namespace testSample
{
    public class MyBarAndDockingController : BarAndDockingController
    {
        public MyBarAndDockingController(IContainer container)
            : base(container)
        {
            
        }
        public MyBarAndDockingController()
        {
         
        }
        protected override void RegisterPaintStyles()
        {
            base.RegisterPaintStyles();
            for (int i = 0; i < PaintStyles.Count; i++)
            {
                PaintStyles[i].ItemInfoCollection.Add(new BarItemInfo("TableBarItem", "Table", 0, typeof(TableBarItem), typeof(BarButtonItemLink), typeof(BarButtonLinkViewInfo), PaintStyles[i].CreateButtonItemLinkPainter(), true, true));
            }
        }
    }
}

File: VB/testSample/TableBarItem/CustomController/MyBarAndDockingController.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraBars
Imports DevExpress.XtraBars.Styles
Imports DevExpress.XtraBars.Controls
Imports DevExpress.XtraEditors
Imports DevExpress.Utils
Imports DevExpress.XtraBars.ViewInfo

Namespace testSample
    Public Class MyBarAndDockingController
        Inherits BarAndDockingController

        Public Sub New(ByVal container As IContainer)
            MyBase.New(container)

        End Sub
        Public Sub New()

        End Sub
        Protected Overrides Sub RegisterPaintStyles()
            MyBase.RegisterPaintStyles()
            For i As Integer = 0 To PaintStyles.Count - 1
                PaintStyles(i).ItemInfoCollection.Add(New BarItemInfo("TableBarItem", "Table", 0, GetType(TableBarItem), GetType(BarButtonItemLink), GetType(BarButtonLinkViewInfo), PaintStyles(i).CreateButtonItemLinkPainter(), True, True))
            Next i
        End Sub
    End Class
End Namespace
